src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -111,7 +112,8 @@ public async Task<bool> TryRefreshAsync()
                 e is KeyVaultReferenceException ||
                 e is RequestFailedException ||
                 ((e as AggregateException)?.InnerExceptions?.All(e => e is RequestFailedException) ?? false) ||
-                e is OperationCanceledException)
+                e is OperationCanceledException ||
+                e.Source.Equals(AzureIdentityExceptionSource, StringComparison.OrdinalIgnoreCase))

```

Comment:
Source can be null right? looks like it could throw a null-ref.

Reply:
If the Source property is null, the .NET runtime automatically sets it to the assembly name in which the exception originated. So I don't expect it to be null for any top-level exceptions. It could be null for inner exceptions. I would still add the null check here just to be safe.